### ISSUE TITLE
STK: Can't link ci_string.cpp.o with Clang using libc++

### ISSUE THREAD

**Issue by dc-snl** [trilinos/Trilinos#11791]
## Bug Report
@alanw0 

### Description
There's a [bug in LLVM libc++](https://github.com/llvm/llvm-project/issues/40908) where `operator<<(ostream, string)` and `operator>>(istream, string)` are declared in `<string>` but defined in `<ostream>` and `<istream>`. This unfortunately bites us when we try to link the object file generated from Trilinos/packages/stk/stk_util/stk_util/util/ci_string.cpp:

```
[2463/4681] Linking CXX shared library packages/stk/stk_util/stk_util/util/libstk_util_util.so.14.1
FAILED: packages/stk/stk_util/stk_util/util/libstk_util_util.so.14.1
: && /tpl/x86_64/openmpi-4.1.5_debug/bin/mpicxx -fPIC -Dbinary_function=__binary_function -Dunary_function=__unary_function -fPIC -O3 -fno-fast-math -Wno-unused-command-line-argument   -DNDEBUG  -Wl,--emit-relocs -static-libstdc++ -static-libgcc -fuse-ld=mold -L/projects/devel/x86_64/compilers/llvm-16.0.1/lib -pthread -ldl -Wl,-rpath,'$ORIGIN/../lib' -L'/scratch/clang-test/install/lib' -shared -Wl,-soname,libstk_util_util.so.14 -o packages/stk/stk_util/stk_util/util/libstk_util_util.so.14.1 [...] packages/stk/stk_util/stk_util/util/CMakeFiles/stk_util_util.dir/ci_string.cpp.o [...] && :
mold: error: undefined symbol: std::__1::basic_ostream<char, std::__1::char_traits<char> >& std::__1::operator<<[abi:v160001]<char, std::__1::char_traits<char>, std::__1::allocator<char> >(std::__1::basic_ostream<char, std::__1::char_traits<char> >&, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&)
>>> referenced by ci_string.cpp
>>>               packages/stk/stk_util/stk_util/util/CMakeFiles/stk_util_util.dir/ci_string.cpp.o:(operator<<(std::__1::basic_ostream<char, std::__1::char_traits<char> >&, std::__1::basic_string<char, ignorecase_traits, std::__1::allocator<char> > const&))
mold: error: undefined symbol: std::__1::basic_istream<char, std::__1::char_traits<char> >& std::__1::operator>>[abi:v160001]<char, std::__1::char_traits<char>, std::__1::allocator<char> >(std::__1::basic_istream<char, std::__1::char_traits<char> >&, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >&)
>>> referenced by ci_string.cpp
>>>               packages/stk/stk_util/stk_util/util/CMakeFiles/stk_util_util.dir/ci_string.cpp.o:(operator>>(std::__1::basic_istream<char, std::__1::char_traits<char> >&, std::__1::basic_string<char, ignorecase_traits, std::__1::allocator<char> >&))
clang-16: error: linker command failed with exit code 1 (use -v to see invocation)
```

I've found a workaround where I patch `Trilinos/packages/stk/stk_util/stk_util/util/ci_string.cpp` to `#include` `<ostream>` and `<istream>` at the beginning. That seems to resolve the issue.

### Steps to Reproduce
1. SHA1:  [398beeb](https://github.com/trilinos/Trilinos/commit/398beeb3a716fff1a32282a3877712ddc1a2ec6c)
1. Configure script: 
```
git clone --branch develop --depth 1 https://github.com/trilinos/Trilinos.git
mkdir build
cd build
cmake \
    -G Ninja \
    -D CMAKE_C_COMPILER=clang \
    -D CMAKE_CXX_COMPILER=clang++ \
    -D BUILD_SHARED_LIBS=ON \
    -D Trilinos_ENABLE_Fortran=OFF \
    -D Trilinos_ENABLE_ALL_PACKAGES=OFF \
    -D Trilinos_ENABLE_ALL_OPTIONAL_PACKAGES=OFF \
    -D Trilinos_ENABLE_STKUtil=ON \
    ../Trilinos
cmake --build . -j40
```
We're using a Clang 16.0.1 build configured with a default `-stdlib` of `libc++`.

1. Configure and Build log: 
[configure_build_log.txt](https://github.com/trilinos/Trilinos/files/11237049/configure_build_log.txt)

**Comment by alanw0** [comment]
@dc-snl thanks for reporting this. This is strange, we have clang builds and haven't seen this before.
I will implement your suggestion and get it committed and pushed.

**Comment by dc-snl** [comment]
Looks like this issue has been resolved as of 926d456. Thanks for looking into this, @alanw0!
